modules/nosql/manager_redis.go

```diff
@@ -131,75 +208,40 @@ func (m *Manager) GetRedisClient(connection string) redis.UniversalClient {
 			fallthrough
 		case "mastername":
 			opts.MasterName = v[0]
-		case "skipverify":
-			fallthrough
-		case "insecureskipverify":
-			insecureSkipVerify, _ := strconv.ParseBool(v[0])
-			tlsConfig.InsecureSkipVerify = insecureSkipVerify
-		case "clientname":
-			client.name = append(client.name, v[0])
+		case "sentinelusername":
+			opts.SentinelUsername = v[0]
+		case "sentinelpassword":
+			opts.SentinelPassword = v[0]
 		}
 	}
 
-	switch uri.Scheme {
-	case "redis+sentinels":
-		fallthrough
-	case "rediss+sentinel":
-		opts.TLSConfig = tlsConfig
-		fallthrough
-	case "redis+sentinel":
-		if uri.Host != "" {
-			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
-		}
-		if uri.Path != "" {
-			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
-				opts.DB = db
-			}
-		}
+	return opts
+}
 
-		client.UniversalClient = redis.NewFailoverClient(opts.Failover())
-	case "redis+clusters":
-		fallthrough
-	case "rediss+cluster":
-		opts.TLSConfig = tlsConfig
-		fallthrough
-	case "redis+cluster":
-		if uri.Host != "" {
-			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
-		}
-		if uri.Path != "" {
-			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
-				opts.DB = db
-			}
-		}
-		client.UniversalClient = redis.NewClusterClient(opts.Cluster())
-	case "redis+socket":
-		simpleOpts := opts.Simple()
-		simpleOpts.Network = "unix"
-		simpleOpts.Addr = path.Join(uri.Host, uri.Path)
-		client.UniversalClient = redis.NewClient(simpleOpts)
-	case "rediss":
-		opts.TLSConfig = tlsConfig
-		fallthrough
-	case "redis":
-		if uri.Host != "" {
-			opts.Addrs = append(opts.Addrs, strings.Split(uri.Host, ",")...)
-		}
-		if uri.Path != "" {
-			if db, err := strconv.Atoi(uri.Path[1:]); err == nil {
-				opts.DB = db
-			}
+// getRedisTlsOptions parses RedisUri TLS configuration parameters and converts them to the go TLS configuration
+// equivalent fields.
+func getRedisTlsOptions(uri *url.URL) *tls.Config {
```

Comment:
TLS options have moved here to their own config.

In keeping with the original behavior of the switch statement, this method checks for both `skipverify` and `insecureskipverify` and if both are present, will parse both (rather than using an `if / else`). But happy to swap to something that eagerly exits if one is found. It's probably not a common edge case that someone provides both _and_ configures them differently expecting it to do something logical.. 😕

Reply:
I think it's unlikely that anyone has both set - so although this is technically a breaking change I think it's probably fine.

Reply:
Yeah glad you agree -- I don't think there's a practical way to ensure the behavior (or if there even was ensured behavior) due to the map iteration 👍🏾